### PR TITLE
Fix integration tests

### PR DESCRIPTION
**Pull request by patrykkulik-microsoft**
This PR fixes the bug in the integration tests and re-enables them. 

First bug is fixed by re-trying validation in our code. For some reason this retry step is done automatically in live tests but not in recordings so doing it explicitly allows us to run tests successfully.

The cnf tests are fixed by running them live only because it is not possible to run the az login command in the recorded tests without credentials. 

I have also modified the recording processors (functions which remove credentials in the recording) because some were missed initially. 
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/aosm/azext_aosm/tests/latest/test_aosm_cnf_publish_and_delete.py [hunk 1]
```diff
@@ -1,91 +1,118 @@
-# --------------------------------------------------------------------------------------------
+# # --------------------------------------------------------------------------------------------
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License. See License.txt in the project root for license information.
+#
+# This is an integration tests for the aosm extension. It tests the following commands for the
```
Comment (Cyclam) [inline src/aosm/azext_aosm/tests/latest/test_aosm_cnf_publish_and_delete.py:5]:
```suggestion
# Integration tests for the aosm extension. They test the following commands for the
```

File: src/aosm/azext_aosm/tests/latest/test_aosm_cnf_publish_and_delete.py [hunk 2]
```diff
@@ -1,91 +1,118 @@
...
+#
 # --------------------------------------------------------------------------------------------
 
-# Currently commented out because of the timeout bug in the testing framework.
+import os
+from typing import Dict
+from azure.cli.testsdk import LiveScenarioTest, ResourceGroupPreparer
+from knack.log import get_logger
+from jinja2 import Template
+from .recording_processors import TokenReplacer, SasUriReplacer
```
Comment (Cyclam) [inline src/aosm/azext_aosm/tests/latest/test_aosm_cnf_publish_and_delete.py:21]: I think you'll need to run isort on this list

File: src/aosm/azext_aosm/tests/latest/test_aosm_cnf_publish_and_delete.py [hunk 3]
```diff
@@ -1,91 +1,118 @@
...
-#     rendered_template = jinja_template.render(**params)
+    return output_path
 
-#     output_path = os.path.join(templates_dir, output_file_name)
 
-#     with open(output_path, "w", encoding="utf-8") as file:
-#         file.write(rendered_template)
+class CnfNsdTest(LiveScenarioTest):
+    """
+    This class contains the integration tests for the aosm extension for cnf definition type.
```
Comment (Cyclam) [inline src/aosm/azext_aosm/tests/latest/test_aosm_cnf_publish_and_delete.py:64]:
```suggestion
    Integration tests for the aosm extension for cnf definition type.
```

File: src/aosm/development.md [hunk 4]
```diff
@@ -142,7 +142,7 @@ python-static-checks fmt
 ```
 
 ### Tests
-The tests in this repository are split into unit tests and integration tests. Both tests live in the `tests/latest` folder and can be run using the `azdev test aosm` command. All tests are expected to pass. All unit tests and Integration tests are run as part of the pipeline. 
+The tests in this repository are split into unit tests and integration tests. Both tests live in the `tests/latest` folder and can be run using the `azdev test aosm` command (you can optionally use the `--live` flag with this command as some integration tests are run only in live mode). All tests are expected to pass. All unit tests and Integration tests are run as part of the pipeline.
```
Comment (Cyclam) [inline src/aosm/development.md:145]:
```suggestion
The tests in this repository are split into unit tests and integration tests. Both tests live in the `tests/latest` folder and can be run using the `azdev test aosm` command (you can optionally use the `--live` flag with this command as some integration tests are run only in live mode, e.g. CNF tests). All tests are expected to pass. All unit tests and Integration tests are run as part of the pipeline. 
```

**Review by Cyclam — APPROVED** [review]
Only minor comments.

I've not reviewed the previously commented out tests, as I assume they've been reviewed before. If that's not the case, and the tests need more detailed review, let m know.

File: src/aosm/azext_aosm/tests/latest/test_aosm_cnf_publish_and_delete.py (hunk 2 above)
Comment (patrykkulik-microsoft) [reply src/aosm/azext_aosm/tests/latest/test_aosm_cnf_publish_and_delete.py:21]: according to my linting they are in the correct order. The recording processors import was redundant so I removed it.

**Pull request merged**